### PR TITLE
Allow tests to override TGW interval, add Compactor job wait property

### PR DESCRIPTION
**Pull request by dlmarion**

### CODE REVIEW

File: server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java [hunk 1]
```diff
@@ -426,8 +426,9 @@ protected void updateCompactionCompleted(TExternalCompactionJob job, TCompaction
    * @throws RetriesExceededException thrown when retries have been exceeded
    */
   protected TExternalCompactionJob getNextJob(Supplier<UUID> uuid) throws RetriesExceededException {
+    final long checkWaitTime = getConfiguration().getTimeInMillis(Property.COMPACTOR_JOB_WAIT_TIME);
     RetryableThriftCall<TExternalCompactionJob> nextJobThriftCall =
-        new RetryableThriftCall<>(1000, RetryableThriftCall.MAX_WAIT_TIME, 0, () -> {
+        new RetryableThriftCall<>(checkWaitTime, checkWaitTime, 0, () -> {
```
Comment (keith-turner) [inline server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java:431]: With this change there is no longer exponential backoff when the queue keeps being empty.  The following would still allow the exponential backoff.

```suggestion
        long maxWaitTime = getConfiguration().getTimeInMillis(Property.COMPACTOR_JOB_WAIT_MAX_TIME);
        long minWaitTime = Math.min(1000, maxWaitTime);
        new RetryableThriftCall<>(minWaitTime, maxWaitTime, 0, () -> {
```

File: core/src/main/java/org/apache/accumulo/core/conf/Property.java [hunk 2]
```diff
@@ -1098,6 +1098,8 @@ public enum Property {
   @Experimental
   COMPACTOR_CLIENTPORT("compactor.port.client", "9133", PropertyType.PORT,
       "The port used for handling client connections on the compactor servers", "2.1.0"),
+  COMPACTOR_JOB_WAIT_TIME("compactor.wait.time.job", "1s", PropertyType.TIMEDURATION,
+      "The amount of time to wait between checks for the next compaction job", "4.0.0"),
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/conf/Property.java:1102]:
```suggestion
  COMPACTOR_JOB_WAIT_MAX_TIME("compactor.wait.time.job", "5m", PropertyType.TIMEDURATION,
      "The maximum amount of time to wait between checks for the next compaction job.  Compactors do exponential backoff when their request for work repeatedly come back empty.  This is the max time the exponential backoff can grow to", "4.0.0"),
```

Comment (dlmarion) [reply core/src/main/java/org/apache/accumulo/core/conf/Property.java:1102]: max wait time property added in 6b5a132

File: core/src/main/java/org/apache/accumulo/core/conf/Property.java [hunk 3]
```diff
@@ -1098,6 +1098,12 @@ public enum Property {
...
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/conf/Property.java:1102]: If we have a max the making this prop a min seems more consistent.

```suggestion
  COMPACTOR_JOB_WAIT_TIME_MIN("compactor.wait.time.job.min", "1s", PropertyType.TIMEDURATION,
      "The minimum amount of time to wait between checks for the next compaction job. Going up to a max of compactor.wait.time.job.max.", "4.0.0"),
```

File: server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java [hunk 4]
```diff
@@ -426,8 +426,12 @@ protected void updateCompactionCompleted(TExternalCompactionJob job, TCompaction
    * @throws RetriesExceededException thrown when retries have been exceeded
    */
   protected TExternalCompactionJob getNextJob(Supplier<UUID> uuid) throws RetriesExceededException {
+    final long startingWaitTime =
+        getConfiguration().getTimeInMillis(Property.COMPACTOR_JOB_WAIT_TIME);
+    final long maxWaitTime =
+        getConfiguration().getTimeInMillis(Property.COMPACTOR_MAX_JOB_WAIT_TIME);
```
Comment (keith-turner) [inline server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java:432]: Not sure if this check is needed, maybe redundant.  Maybe RetryableThriftCall or Retry is already doing this check. 

```suggestion
        getConfiguration().getTimeInMillis(Property.COMPACTOR_MAX_JOB_WAIT_TIME);
     Preconditions.checkArgument(   startingWaitTime <= maxWaitTime,  "TODO Message including props names");
```

**Review by keith-turner — APPROVED** [review]

File: server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java (hunk 4 above)
Comment (dlmarion) [reply server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java:432]: It is being done in `Retry.maxWait`.

File: core/src/main/java/org/apache/accumulo/core/conf/Property.java (hunk 3 above)
Comment (dlmarion) [reply core/src/main/java/org/apache/accumulo/core/conf/Property.java]: Renamed property and updated description in 66fc591.

**Pull request merged**